### DISCUSSION TITLE
Why is DJANGO_SETTINGS_MODULE="config.settings.test" in the compose/production/django/Dockerfile?

### DISCUSSION THREAD

**Discussion by natehouk** [cookiecutter/cookiecutter-django#4524] [Q&A]
What is the purpose of the following code in the file compose/production/django/Dockerfile (line 100)?

```
RUN DATABASE_URL="" \
  CELERY_BROKER_URL="" \
  DJANGO_SETTINGS_MODULE="config.settings.test" \
  python manage.py compilemessages
```

Isn't this dangerous because you are using "config.settings.test" in the production environment? And why does DATABASE_URL get set to ""?

Are these actually overriding the production settings? Or am I misunderstanding and this is some attempt to cleanse the environment variables on the production server?

Shouldn't this entire line be removed from the production django compose file?

Please enlighten me.

**Comment by browniebroke** [comment]
This is done to run `compilemessages` while the Docker image is built and was added in https://github.com/cookiecutter/cookiecutter-django/pull/4363. The command compiles the `.po` files into `.mo` files, as explained in [the translations README](https://github.com/cookiecutter/cookiecutter-django/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/locale/README.md?plain=1). We need to run this at build time because it doesn't work part that point (https://github.com/cookiecutter/cookiecutter-django/issues/4189). 

We don't really need a value for `DATABASE_URL` or `CELERY_BROKER_URL` (or most settings) to compile the translation strings, and the production settings aren't usable at build time since most of the env variables will be unset. Therefore, we set these env variables just for this command, at build time, and it shouldn't affect what setting file is used at runtime on prod. Using the test setting was the simpler way to run this command in that context. 

I'm open to a better solution, if you have one to suggest.
